### PR TITLE
Refactor config api and split config form

### PR DESCRIPTION
**Pull request by leehuwuj**

### CODE REVIEW

File: src/routers/management/files.py [hunk 1]
```diff
@@ -40,13 +40,8 @@ def remove_file(file_name: str):
...
         FileHandler.remove_file(file_name)
     except FileNotFoundError:
-        return JSONResponse(
-            status_code=404,
-            content={
-                "error": "FileNotFoundError",
-                "message": f"File {file_name} not found.",
-            },
-        )
+        # Ignore the error if the file is not found
```
Comment (marcusschiesser) [inline src/routers/management/files.py:43]: comment why we do this

File: src/models/chat_config.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+from typing import List, Self
+from pydantic import Field, validator, field_validator
+from pydantic.json_schema import CoreSchema
+from src.models.base_env import BaseEnvConfig
+
+
+class NewlineListEnv(List[str]):
```
Comment (marcusschiesser) [inline src/models/chat_config.py:7]: useful for other configs? maybe move to base?

File: src/routers/management/config.py [hunk 3]
```diff
@@ -43,7 +78,7 @@ def update_config(
     )
 
 
-@r.get("/models")
+@r.get("/available_models", tags=["Model config"])
```
Comment (marcusschiesser) [inline src/routers/management/config.py:81]: part of models group? e.g.:
```suggestion
@r.get("/models/list", tags=["Model config"])
```

File: admin/sections/config/model.tsx [hunk 4]
```diff
@@ -59,55 +93,57 @@ export const ModelConfig = ({
...
   return (
     <ExpandableSection
       name="update-model"
-      title={values.configured ? "Update model" : "Start"}
-      description={
-        values.configured
-          ? "Change to a different model or use another provider"
-          : "Set up an AI model to start the app."
-      }
+      title={"Update model"}
```
Comment (marcusschiesser) [inline admin/sections/config/model.tsx:101]: what about showing different text depending on `values.configured`?

Comment (leehuwuj) [reply admin/sections/config/model.tsx]: Ah great catch! i forgot to add this back :D

File: admin/pages/index.tsx [hunk 5]
```diff
@@ -62,7 +62,10 @@ export default function Home() {
                 "m-auto": !configured,
               })}
             >
-              <ModelConfig setConfigured={setConfigured} />
+              <ModelConfig
+                configured={configured ?? false}
+                setConfigured={setConfigured}
+              />
```
Comment (marcusschiesser) [inline admin/pages/index.tsx:68]:
```suggestion
              <ModelConfig
                 title={configured ? "Update model" : "Start"}
      description={
        configured
          ? "Change to a different model or use another provider"
          : "Set up an AI model to start the app."
      }
                onConfigChange={handleModelConfigChange}
              />
```

and in `handleModelConfigChange` you retrigger `fetchIsAppConfigured()`

**Pull request merged**